csrc/quantization/fp8/amd/gemm_kernel.cu

```diff
@@ -0,0 +1,164 @@
+  } else {
+    hipblas_out_type = HIP_R_16F;
+  }
+
+  constexpr bool transpose_result = true;
+  bool transpose_a;
+  bool transpose_b;
+  if ((b_strides[0] == 1) &&
+      (b_strides[1] >= std::max<int64_t>(1, b_sizes[0]))) {
+    transpose_b = false;
+  } else if ((b_strides[1] == 1) &&
+             (b_strides[0] >= std::max<int64_t>(1, b_sizes[1]))) {
+    transpose_b = true;
+  } else {
+    assert(false &&
+           "unusual strides detected, may need to clone a contiguous tensor");
+  }
+  if ((a_strides[0] == 1) &&
+      (a_strides[1] >= std::max<int64_t>(1, a_sizes[0]))) {
+    transpose_a = false;
+  } else if ((a_strides[1] == 1) &&
+             (a_strides[0] >= std::max<int64_t>(1, a_sizes[1]))) {
+    transpose_a = true;
+  } else {
+    assert(false &&
+           "unusual strides detected, may need to clone a contiguous tensor");
+  }
+
+  if (transpose_result) {
+    bool tmp = transpose_a;
+    transpose_a = !transpose_b;
+    transpose_b = !tmp;
+    a_strides = b.strides();
+    b_strides = a.strides();
+    a_sizes = b.sizes();
+    b_sizes = a.sizes();
+  }
+
+  float alpha = 1.0f;
+  float beta = 0.0f;
+  int64_t m = a_sizes[transpose_result ? 1 : 0];
+  int64_t k = a_sizes[transpose_result ? 0 : 1];
+  int64_t n = b_sizes[transpose_result ? 0 : 1];
+
+  void* d_a = static_cast<void*>((transpose_result ? b : a).data_ptr());
+  void* d_b = static_cast<void*>((transpose_result ? a : b).data_ptr());
+  void* d_d = static_cast<void*>(result.data_ptr());
+
+  auto d_scale_a = transpose_result ? scale_a.data_ptr() : scale_a.data_ptr();
+  auto d_scale_b = transpose_result ? scale_b.data_ptr() : scale_b.data_ptr();
+  auto d_scale_d =
+      scale_result.has_value() ? scale_result.value().data_ptr() : nullptr;
+
+  auto handle = at::cuda::getCurrentCUDABlasLtHandle();
+  auto stream = at::cuda::getCurrentCUDAStream();
+
+  hipblaslt_ext::GemmPreference gemmPref;
+  gemmPref.setMaxWorkspaceBytes(0);
+  hipblaslt_ext::Gemm gemm(handle, transpose_a ? HIPBLAS_OP_T : HIPBLAS_OP_N,
+                           transpose_b ? HIPBLAS_OP_T : HIPBLAS_OP_N,
+                           HIP_R_8F_E4M3_FNUZ, HIP_R_8F_E4M3_FNUZ,
+                           hipblas_out_type, hipblas_out_type,
+                           HIPBLAS_COMPUTE_32F);
+
+  // TODO(HaiShaw): Add Epilogue usage in cases to support Bias, etc.
+  hipblaslt_ext::GemmEpilogue
+      epilogue{};  // No action needed, default is HIPBLASLT_EPILOGUE_DEFAULT.
+                   // (Gemm only)
+  hipblaslt_ext::GemmInputs inputs;
+  inputs.a = d_a;
+  inputs.b = d_b;
+  inputs.c = d_d;
+  inputs.d = d_d;
+  inputs.alpha = &alpha;
+  inputs.beta = &beta;
+  inputs.scaleA = d_scale_a;
+  inputs.scaleB = d_scale_b;
+  inputs.scaleD = d_scale_d;
+
+  auto&& problem = gemm.getProblemTypes();
+  auto lda = problem.op_a == HIPBLAS_OP_N ? m : k;
+  auto ldb = problem.op_b == HIPBLAS_OP_N ? k : n;
+  auto ldc = m;
+  auto strideA = m * k;
+  auto strideB = n * k;
+  auto strideC = m * n;
```

Comment:
Here you are constructing the strides using the shape of the operands. This is usually an OK thing to do but the kernel will break if passed in subtensors. Could you add some `TORCH_CHECK`s to guard against this? (e.g. `TORCH_CHECK(a.is_contiguous() && b.is_contiguous())` -- you may need to deal with transposes though.

Alternatively, it would be even better to support subtensors, and to add a unit test to check this case